### PR TITLE
ENG-12821: Updates to describe ipywidgets

### PR DESCRIPTION
**Pull request by bredamc**
ENG-12821: Updates to describe ipywidgets

### CODE REVIEW

**Review by ChristianZaccaria — COMMENTED** [review]
Looks amazing, thanks Breda!

Two things to mention:

- Something else that we could add (not necessary), is that the `view_clusters()` lists all the RayClusters in the user's namespace/project. However, an admin can explicitly provide a namespace as an argument i.e.,

  ```
  view_clusters("project-1-namespace")
  ```
  This would list the RayClusters in that namespace for the admin. For users, this would likely fail as they would only have access to their own project's namespace.

- We have this [3_widget_example.ipynb](https://github.com/project-codeflare/codeflare-sdk/blob/main/demo-notebooks/guided-demos/3_widget_example.ipynb) notebook where we will be adding all new widgets to. - This is a notebook to simply show the user all the UI widgets that are available for them.

**Review by ChristianZaccaria — COMMENTED** [review]
/lgtm thank you, awesome work!

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+endif::[]
+ifdef::upstream[]
+. Run all of the demo notebooks in the order indicated by the file-name prefix (`0_`, `1_`, and so on), as described in link:{odhdocshome}/working-with-distributed-workloads/#running-the-demo-notebooks-from-the-codeflare-sdk_distributed-workloads[Running the demo notebooks from the CodeFlare SDK].
+endif::[]
+
+. In each demo notebook, when you run the cluster configuration step, the following interactive controls are automatically shown in the notebook:
+
+* *Cluster Up*: You can click this button to start the Ray cluster. 
+This button is equivalent to the `cluster.up()` command. 
+A message is displayed to indicate whether the cluster was successfully created or not.
```
Comment (eturner24) [inline modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:78]:
```suggestion
A message is displayed that indicates whether the cluster was successfully created.
```
Minor nitpick - "whether" implies "or not"

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+* *Cluster Down*: You can click this button to delete the Ray cluster. 
+This button is equivalent to the `cluster.down()` command.
+The cluster is deleted immediately, you are not prompted to confirm the deletion.
```
Comment (eturner24) [inline modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:82]:
```suggestion
The cluster is deleted immediately. You are not prompted to confirm the deletion.
```
or
```suggestion
The cluster is deleted immediately; you are not prompted to confirm the deletion.
```

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+A message is displayed to indicate whether the cluster was successfully deleted or not.
```
Comment (eturner24) [inline modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:83]:
```suggestion
A message is displayed that indicates whether the cluster was successfully deleted.
```

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
++
+Under this heading, a toggle button is shown for each existing cluster. 
+Click a cluster name to select the cluster.
+The cluster details section is updated to show details about the selected cluster; for example, cluster name, {productname-short} project name, cluster resource information, and cluster status.
+
+* *Delete cluster*
++
+Click this button to delete the selected cluster.
+This button is equivalent to the *Cluster Down* button.
+The cluster is deleted immediately, you are not prompted to confirm the deletion.
```
Comment (eturner24) [inline modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:160]: Same feedback as previous

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+A message is shown to confirm the deletion, and the corresponding button is no longer shown under the *Select an existing cluster* heading.
```
Comment (eturner24) [inline modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:161]: > A message is shown to confirm the deletion,

Doesn't this contradict the previous sentence?

File: modules/running-the-demo-notebooks-from-the-codeflare-sdk.adoc [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+The default images are AMD64 images, which might not work on other architectures. 
+
+Additional ROCm-compatible Ray cluster images are available as Developer Preview software.
+These images are compatible with AMD accelerators that are supported by ROCm 6.1. 
+These images are AMD64 images, which might not work on other architectures.
+
+[IMPORTANT]
+====
+The ROCm-compatible Ray cluster images are Developer Preview software only. 
+Developer Preview software is not supported by Red{nbsp}Hat in any way and is not functionally complete or production-ready.
```
Comment (eturner24) [inline modules/running-the-demo-notebooks-from-the-codeflare-sdk.adoc:144]: Should we use an attribute for the company name here?

File: modules/running-the-demo-notebooks-from-the-codeflare-sdk.adoc [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+.. Optional: Assign a dictionary of `labels` parameters to the Ray cluster for identification and management purposes, as shown in the following example:
++
+.Example labels assignment
+[source,bash,subs="+quotes"]
+----
+labels = {"exampleLabel1": "exampleLabel1Value", "exampleLabel2": "exampleLabel2Value"}
+----
+
+. In the `2_basic_interactive.ipynb` notebook, ensure that the following Ray cluster authentication code is included after the Ray cluster creation section.
```
Comment (eturner24) [inline modules/running-the-demo-notebooks-from-the-codeflare-sdk.adoc:175]:
```suggestion
. In the `2_basic_interactive.ipynb` notebook, ensure that the following Ray cluster authentication code is included after the Ray cluster creation section:
```

**Review by eturner24 — CHANGES_REQUESTED** [review]

File: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+* You can access a data science cluster that is configured to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/managing_openshift_ai/managing_distributed_workloads[Managing distributed workloads].
+endif::[]
+ifdef::upstream[]
+* You can access a data science cluster that is configured to run distributed workloads as described in link:{odhdocshome}/managing-odh/#_managing_distributed_workloads[Managing distributed workloads].
+endif::[]
+
+* You can access the following software from your data science cluster:
+** A Ray cluster image that is compatible with your hardware architecture
+** The data sets and models to be used by the workload
+** The Python dependencies for the workload, either in a Ray image or in your own Python Package Index (PyPI) server
```
[the remark this answers is not in the document]
Comment (bredamc) [reply modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc:40]: These lines are for a list that is at a different level than the list items in lines 17 and 18. The items in this second-level list are all fragments, so no period needed, according to _IBM Style_.

**Pull request merged**